tests/unit/app_integrations/test_apps/test_app_base.py

```diff
@@ -212,3 +212,18 @@ def test_gather_running(self, finalize_mock):
         self._app._config['current_state'] = 'running'
         self._app.gather()
         finalize_mock.assert_not_called()
+
+    @patch('requests.get')
+    def test_make_request_bad_response(self, requests_mock):
+        """App Integration - Make Request, Bad Response"""
+        requests_mock.return_value = Mock(
+            status_code=404,
+            json=Mock(side_effect=[{'message': 'something went wrong'}])
+        )
```

Comment:
Nice! I know this is copied, but I didn't know you could use a raw Mock like this